aws instance-connect panics when ssh connection problems exist

**Describe the bug**
I couldn't connect via ssh to my EC2 instance, which I didn't know before executing cnquery:
```
cnquery run aws ec2 instance-connect ec2-user@3.125.33.86 --region us-central-1a -c 'asset.name'                                                                          
! CLI pre-processing encountered an issue error="unknown flag: --region"
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
! your platform is not supported for hostname detection
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x127637b]

goroutine 52 [running]:
go.mondoo.com/cnquery/providers/os/connection.(*SshConnection).PlatformID(...)
	/home/christian/workspace/mondoo/github.com/cnquery/providers/os/connection/ssh.go:346
go.mondoo.com/cnquery/providers/os/id/sshhostkey.Detect({0x1f76970?, 0x0?}, 0xc0001f9080?)
	/home/christian/workspace/mondoo/github.com/cnquery/providers/os/id/sshhostkey/sshhostkey.go:20 +0xbb
go.mondoo.com/cnquery/providers/os/provider.GatherPlatformInfo({0x1f76970?, 0x0?}, 0xc0005677e0?, {0x1c97c67?, 0x0?})
	/home/christian/workspace/mondoo/github.com/cnquery/providers/os/provider/platform.go:178 +0x77b
go.mondoo.com/cnquery/providers/os/provider.IdentifyPlatform({0x1f76970, 0x0}, 0xc000306fc0?, {0xc00078b5c0, 0x3, 0x3})
	/home/christian/workspace/mondoo/github.com/cnquery/providers/os/provider/platform.go:52 +0x15a
go.mondoo.com/cnquery/providers/os/provider.(*Service).connect(0xc0003193b0, 0xc0000b66c0, {0x1f69fd0?, 0xc0004902c0})
	/home/christian/workspace/mondoo/github.com/cnquery/providers/os/provider/provider.go:302 +0x14a
...
```

The EC2 instance couldn't be reached.

**To Reproduce**
Steps to reproduce the behavior:

1. Create unreachable VM/ or use wrong IP
2. Run cnquery
3. Note the error

**Expected behavior**
We shouldn't panic, but show an error that we couldn't connect.

--- a/providers/os/provider/provider.go
+++ b/providers/os/provider/provider.go
@@ -300,6 +300,9 @@ func (s *Service) connect(req *plugin.ConnectReq, callback plugin.ProviderCallba
 	case SshConnectionType:
 		s.lastConnectionID++
 		conn, err = connection.NewSshConnection(s.lastConnectionID, conf, asset)
+		if err != nil {
+			return nil, err
+		}
 		fingerprint, err := IdentifyPlatform(conn, asset.Platform, []string{ids.IdDetector_Hostname, ids.IdDetector_CloudDetect, ids.IdDetector_SshHostkey})
 		if err == nil {
 			if conn.Asset().Connections[0].Runtime != "vagrant" {
@@ -311,6 +314,9 @@ func (s *Service) connect(req *plugin.ConnectReq, callback plugin.ProviderCallba
 	case TarConnectionType:
 		s.lastConnectionID++
 		conn, err = connection.NewTarConnection(s.lastConnectionID, conf, asset)
+		if err != nil {
+			return nil, err
+		}
 		fingerprint, err := IdentifyPlatform(conn, asset.Platform, []string{ids.IdDetector_Hostname})
 		if err == nil {
 			asset.Name = fingerprint.Name
@@ -320,6 +326,9 @@ func (s *Service) connect(req *plugin.ConnectReq, callback plugin.ProviderCallba
 	case DockerSnapshotConnectionType:
 		s.lastConnectionID++
 		conn, err = connection.NewDockerSnapshotConnection(s.lastConnectionID, conf, asset)
+		if err != nil {
+			return nil, err
+		}
 		fingerprint, err := IdentifyPlatform(conn, asset.Platform, []string{ids.IdDetector_Hostname})
 		if err == nil {
 			asset.Name = fingerprint.Name
@@ -358,6 +367,9 @@ func (s *Service) connect(req *plugin.ConnectReq, callback plugin.ProviderCallba
 	case FilesystemConnectionType:
 		s.lastConnectionID++
 		conn, err = connection.NewFileSystemConnection(s.lastConnectionID, conf, asset)
+		if err != nil {
+			return nil, err
+		}
 		fingerprint, err := IdentifyPlatform(conn, asset.Platform, []string{ids.IdDetector_Hostname})
 		if err == nil {
 			asset.Name = fingerprint.Name

